roles/testnode/tasks/apt_systems.yml

```diff
@@ -32,6 +32,7 @@
   args:
     creates: /etc/fstab.bak
   register: add_user_xattr
+  when: modify_fstab is defined
```

Comment:
modify_fstab will always be defined because it's in defaults/main.yml, so this condition would not make this skip if modify_fstab is false.  All you need here is, 'when: modify_fstab'.